src/internal/ajax/ajax.ts

```diff
@@ -295,6 +295,21 @@ export function fromAjax<T>(config: AjaxConfig): Observable<AjaxResponse<T>> {
       headers['x-requested-with'] = 'XMLHttpRequest';
     }
 
+    // Allow users to provide their XSRF cookie name and the name of a custom header to use to
+    // send the cookie. This
+    if (config.withCredentials && config.xsrfCookieName && config.xsrfHeaderName) {
+      try {
+        const match = document.cookie.match(new RegExp(`(^|;\\s*)(${name})=([^;]*)`));
```

Comment:
Instead of the try/catch here, we should check to see if `document` exists at all. No reason to throw really, we can probably just noop this if it happens to be a non-browser environment. As long as whatever we're doing aligns with contemporaries like axios, et al.

We can put a cookie-getter function together based off of the function [here](https://stackoverflow.com/a/25490531/135786)

And do something like:

```ts
const { withCredentials, xsrfCookieName, xsrfHeaderName } = config;
if (withCredentials && xsrfCookiename && xsrfHeaderName) {
  headers[xsrfHeadername] = getCookie(xsrfCookiename);
}
```

and below something like:

```ts
function getCookie(name: string): string {
  if (typeof document !== 'undefined') {
    return document.cookie.match(`(^|;)\\s*${name}\\s*=\\s*([^;]+)`)?.pop() ?? '';
  }
}
```

Then we can ditch the error handling below.

Otherwise, we'll be introducing a non-isometric behavior that actually causes errors at runtime. It's still not isometric, but at least it won't error and it's relatively understandable.

Reply:
> We can put a cookie-getter function together

Done.

> Then we can ditch the error handling

Actually your function would still throw a `ReferenceError` in Node. This can be avoided by optional chaining the `globalThis` object!

```javascript
globalThis?.document.cookie.match(...)?.pop() ?? '';
```

Reply:
Another option is we just rely on `config.crossDomain` instead of determining if the request is the same origin at runtime, whichever you prefer sgtm.

Reply:
Turns out `globalThis` is not supported in Node 11, so in order to get all tests to pass we have to do some checking for `ReferenceError`s 😢